ivy/functional/ivy/statistical.py

```diff
@@ -1148,6 +1148,29 @@ def cumprod(
     )
 
 
+@handle_array_function
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_array_like_without_promotion
+@handle_nestable
+@handle_exceptions
+def cummin(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    axis: int = 0,
+    reverse: bool = False,
+    dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    TODO: add docs
```

Comment:
We should be adding the docs before merging :)

Reply:
apologies. I was so focused on getting my implementation right, I totally forgot about the docstrings.